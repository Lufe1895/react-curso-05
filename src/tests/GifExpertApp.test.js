const { shallow } = require("enzyme")
import GifExpertApp from '../GifExpertApp';

describe('Pruebas en <GifExpertApp />', () => {
    test('Debe mostrarse correctamente', () => {
        const wrapper = shallow(<GifExpertApp />);
        expect(wrapper).toMatchSnapshot();
    });

    test('Debe mostrar una lista de categorÃ­as', () => {
        const categories = ['Hellblade', 'Darksiders 3'];
        const wrapper = shallow(<GifExpertApp defaultCategories={categories} />);
    
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('GifGrid').length).toBe(categories.length);
    });
    
});
