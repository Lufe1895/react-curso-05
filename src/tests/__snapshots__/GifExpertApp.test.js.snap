// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifExpertApp /> Debe mostrar una lista de categor√≠as 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="Hellblade"
      key="Hellblade"
    />
    <GifGrid
      category="Darksiders 3"
      key="Darksiders 3"
    />
  </ol>
</Fragment>
`;

exports[`Pruebas en <GifExpertApp /> Debe mostrarse correctamente 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;
